src/java/org/apache/cassandra/utils/StorageCompatibilityMode.java

```diff
@@ -35,6 +35,13 @@ public enum StorageCompatibilityMode
      */
     CASSANDRA_4(4),
 
+    /**
+     * Similar to CASSANDRA_4.
+     * The new features in 6.0 are
+     * - ZSTD dictionary-based compression. Once SSTables are compressed with dictionary, they cannot be rolled back.
+     */
+    CASSANDRA_5(5),
+
```

Comment:
Is it necessary? `StorageCompatibilityMode` is to check the sstable format compatibility, which is not changed in this patch. Also, the dictionary feature is opt-in. I'd incline removing the code. It seems unnecessary. 

cc: @smiklosovic , @rustyrazorblade